python/lsst/fgcmcal/fgcmBuildStarsTable.py

```diff
@@ -0,0 +1,347 @@
+        sourceSelector.doUnresolved = True
+        sourceSelector.doSignalToNoise = True
+        sourceSelector.doIsolated = True
+
+        sourceSelector.isolated.parentName = 'parentSourceId'
+        sourceSelector.isolated.nChildName = 'Deblend_nChild'
+
+        sourceSelector.signalToNoise.fluxField = self.instFluxField
+        sourceSelector.signalToNoise.errField = self.instFluxField + 'Err'
+        sourceSelector.signalToNoise.minimum = 10.0
+        sourceSelector.signalToNoise.maximum = 1000.0
+
+        sourceSelector.unresolved.name = 'extendedness'
+        sourceSelector.unresolved.maximum = 0.5
+
+
+class FgcmBuildStarsTableTask(FgcmBuildStarsBaseTask):
+    """
+    Build stars for the FGCM global calibration, using sourceTable_visit catalogs.
+    """
+    ConfigClass = FgcmBuildStarsTableConfig
+    RunnerClass = FgcmBuildStarsRunner
+    _DefaultName = "fgcmBuildStarsTable"
+
+    @classmethod
+    def _makeArgumentParser(cls):
+        """Create an argument parser"""
+        parser = pipeBase.ArgumentParser(name=cls._DefaultName)
+        parser.add_id_argument("--id", "sourceTable_visit", help="Data ID, e.g. --id visit=6789")
+
+        return parser
+
+    def findAndGroupDataRefs(self, butler, dataRefs):
+        """
+        Find and group dataRefs (by visit).
+
+        Parameters
+        ----------
+        butler: `lsst.daf.persistence.Butler`
+        dataRefs: `list` of `lsst.daf.persistence.ButlerDataRef`
+           Data references for the input visits.
+
+        Returns
+        -------
+        groupedDataRefs: `dict`
+           Dictionary with visit keys, and `list`s of `lsst.daf.persistence.ButlerDataRef`
+        """
+
+        self.log.info("Grouping dataRefs by %s" % (self.config.visitDataRefName))
+
+        camera = butler.get('camera')
+
+        ccdIds = []
+        for detector in camera:
+            ccdIds.append(detector.getId())
+        # Insert our referenceCCD first:
+        ccdIds.insert(0, self.config.referenceCCD)
+
+        # The visitTable building code expects a dictionary of groupedDataRefs
+        # keyed by visit, the first element as the "primary" calexp dataRef.
+        # We then append the sourceTable_visit dataRef at the end for the
+        # code which does the data reading (fgcmMakeAllStarObservations).
+
+        groupedDataRefs = {}
+        for dataRef in dataRefs:
+            visit = dataRef.dataId[self.config.visitDataRefName]
+
+            groupedDataRefs[visit] = []
+
+            # Find an existing calexp (we need for psf and metadata)
+            # and make the relevant dataRef
+            for ccdId in ccdIds:
+                calexpRef = butler.dataRef('calexp', dataId={self.config.visitDataRefName: visit,
+                                                             self.config.ccdDataRefName: ccdId})
+                if calexpRef.datasetExists():
+                    groupedDataRefs[visit].append(calexpRef)
+                    break
```

Comment:
The psf varies across the focal plane, yes.  However, the psf size does not need to be used in detail, only a rough scaling is derived from it and per-visit is fine.  Although getting the psf size per ccd would be even better, the speed degradation to get the psf size per ccd per visit is unacceptably slow (like, it takes CPU-weeks).  We really need these metadata to be collated into a database somewhere because we know the size when we do the psf modeling, we just don't store it.  This is hopefully something that gen3 will help?

Reply:
I think this is a good question for #dm-science-pipelines, and should probably result in a ticket. We generally have a policy of not storing things you can compute from the existing objects, but when that computation takes forever to get a simple number, we should probably revise that. Whether that goes into the gen3 database, or some other metadata store I don't have an opinion on.

Reply:
Were you able to follow up on this on Slack? I think this touches on some broader policy questions (what do we compute, when) that I've been curious about in jointcal too.